go/adbc/go.mod

```diff
@@ -17,7 +17,9 @@
 
 module github.com/apache/arrow-adbc/go/adbc
 
-go 1.21
+go 1.22.0
+
+toolchain go1.22.7
```

Comment:
we should probably change these to `go1.22` and `toolchain go1.22.4` so that we aren't subject to the issue we filed on the go runtime which was introduced with `go1.22.5` until they fix it.